USFMToolsSharp.Renderers.Docx/DocxRenderer.cs

```diff
@@ -198,30 +194,29 @@ private void RenderMarker(Marker input, XWPFParagraph parentParagraph = null, bo
                     break;
             }
         }
-        private void RenderFootnotes()
+        private void RenderFootnotes(StyleConfig styles)
         {
-
-            if (FootnoteTextTags.Count > 0)
+            if (FootnoteMarkers.Count > 0)
             {
-                XWPFParagraph renderFootnoteHeader = newDoc.CreateParagraph();
-                XWPFRun FootnoteHeader = renderFootnoteHeader.CreateRun();
-                FootnoteHeader.SetText("Footnotes");
-                FootnoteHeader.FontSize = 24;
+                XWPFParagraph renderFootnoteStart = newDoc.CreateParagraph();
+                renderFootnoteStart.BorderTop = Borders.Single;
 
-                foreach(KeyValuePair<string,Marker> footnoteKVP in FootnoteTextTags)
+                foreach (KeyValuePair<string,Marker> footnoteKVP in FootnoteMarkers)
                 {
                     XWPFParagraph renderFootnote = newDoc.CreateParagraph();
                     XWPFRun footnoteMarker = renderFootnote.CreateRun();
                     footnoteMarker.SetText(footnoteKVP.Key);
                     footnoteMarker.Subscript = VerticalAlign.SUPERSCRIPT;
 
-                    foreach(Marker marker in footnoteKVP.Value.Contents)
+                    StyleConfig markerStyle = (StyleConfig)styles.Clone();
```

Comment:
This should be outside of the foreach loop. Since it doesn't change at all.

Reply:
@rbnswartz Clone() is now outside of the loop. No unnecessary cloning.